core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -422,6 +425,12 @@ private[spark] class MesosClusterScheduler(
       "--driver-cores", desc.cores.toString,
       "--driver-memory", s"${desc.mem}M")
 
+    if (desc.schedulerProperties.get("spark.jars").isDefined) {
+      val jarFiles = desc.schedulerProperties.get("spark.jars").get
+        .split(",").map(uri => uri.substring(uri.lastIndexOf('/') + 1)).mkString(",")
+      options ++= Seq("--jars", jarFiles)
+    }
+
```

Comment:
Looking at code further in this function, it looks like the right construct might be `map`? The result of `get` is an `Option`, so `map` will do nothing if it's not there. This is equivalent, I think.